src/model/generateRestHandlers.ts

```diff
@@ -159,7 +160,7 @@ export function generateRestHandlers<
           const id = req.params[primaryKey]
           const where: WeakQuerySelectorWhere<typeof primaryKey> = {
             [primaryKey]: {
-              equals: id,
+              equals: typeof primaryKeyValue === 'number' ? +id : id as any,
```

Comment:
What is the type error here so we cast the `id` value to `any`?

Reply:
> Type '{ [x: string]: { equals: string | number; }; }' is not assignable to type 'WeakQuerySelectorWhere<PrimaryKeyType>'.

![CleanShot 2021-11-01 at 11 53 05](https://user-images.githubusercontent.com/7356098/139667524-77b3cb1c-2904-4835-b0f7-59959bbe3ba6.png)

Reply:
It seems like `WeakQuerySelectorWhere<typeof primaryKey>` is an incorrect type, specifically, the generic is incorrect. `typeof primaryKey` will always be `string` because `primaryKey` holds the primary key property's _name_.

Instead, it should be `WeakQuerySelectorWhere<PrimaryKeyType>`, indicating that `id` is a union of all supported primary key types.

Reply:
It's the same thing. `typeof primaryKey` is `PrimaryKeyType`.

![CleanShot 2021-11-01 at 12 24 07](https://user-images.githubusercontent.com/7356098/139670906-22df4e09-b0c7-41fe-88c5-4dd911dac46e.png)

Reply:
I've cast `id as string` so we at least don't resort to `any`.